client/src/uri_util.ts

```diff
@@ -13,42 +13,49 @@ import {
 import { AppState } from './store';
+import { CHARACTER_MAP, LABEL_MAP, STAGE_MAP } from './types';
 
 // export const PARAM_TO_CLIENT_SORT: { [key: string]: string } = { [queryParams.SORT_PARAM_DATE]: SORT_DATE, [queryParams.SORT_PARAM_SCORE]: SORT_SCORE };
 // export const CLIENT_SORT_TO_PARAM: { [key: string]: string } = { [SORT_DATE]: queryParams.SORT_PARAM_DATE, [SORT_SCORE]: queryParams.SORT_PARAM_SCORE };
 
-// export const getFilters = (queryString: string) => _.pick(
-//   getDisplayQueryParams(queryString),
-//   ['currentMainChars', 'currentVsChars', 'currentStages', 'currentStandaloneTags'],
-// );
-
-// const getDisplayQueryParams = (query: string) => {
-//   const queryMap = URI(query).query(true) as { [key: string]: string };
-//   return {
-//     ...queryMap[queryParams.QUERY_SORT] && { currentSort: getSort(query) },
-//     ...queryMap[queryParams.QUERY_LIMIT] && { currentPageSize: getPageSize(query) },
-//     ...queryMap[queryParams.QUERY_OFFSET] && { currentOffset: getOffset(query) },
-//     ...queryMap[queryParams.QUERY_MAIN_CHARS] && { currentMainChars: getCharFilters(queryMap[queryParams.QUERY_MAIN_CHARS]) },
-//     ...queryMap[queryParams.QUERY_VS_CHARS] && { currentVsChars: getCharFilters(queryMap[queryParams.QUERY_VS_CHARS]) },
-//     ...queryMap[queryParams.QUERY_STAGES] && { currentStages: getStageFilters(queryMap[queryParams.QUERY_STAGES]) },
-//     ...queryMap[queryParams.QUERY_TAGS] && { currentStandaloneTags: getTagFilters(queryMap[queryParams.QUERY_TAGS]) },
-//   };
-// };
+const queryStringToObject = (query: string) =>
+  URI(query).query(true) as { [key: string]: string };
 
-// export const getSort = (query: string) => PARAM_TO_CLIENT_SORT[(URI(query).query(true) as { [key: string]: string })[queryParams.QUERY_SORT]];
+const paramStringToFilters = <V>(param: string, filterMap: Map<string, V>) =>
+  new Set(
+    param
+      .split(',')
+      .map((id) => filterMap.get(id))
+      .filter(isPresent),
+  );
 
-export const getPageSize = (query: string) =>
-  parseInt(
-    (URI(query).query(true) as { [key: string]: string })[QUERY_LIMIT],
-    10,
+export const getMainCharFilters = (query: string) =>
+  paramStringToFilters(
+    queryStringToObject(query)[QUERY_MAIN_CHARS],
+    CHARACTER_MAP,
   );
 
-export const getOffset = (query: string) =>
-  parseInt(
-    (URI(query).query(true) as { [key: string]: string })[QUERY_OFFSET],
-    10,
+export const getVsCharFilters = (query: string) =>
+  paramStringToFilters(
+    queryStringToObject(query)[QUERY_VS_CHARS],
+    CHARACTER_MAP,
   );
 
+export const getStageFilters = (query: string) =>
+  paramStringToFilters(queryStringToObject(query)[QUERY_STAGES], STAGE_MAP);
+
+export const getTagFilters = (query: string) =>
+  paramStringToFilters(queryStringToObject(query)[QUERY_TAGS], LABEL_MAP);
+
+export const getSort = (query: string) =>
+  queryStringToObject(query)[QUERY_SORT];
+
+export const getPageSize = (query: string) =>
```

Comment:
The method where we parse the URL for a page size doesn't seem like the right place to put the page size default.

Reply:
So this actually isn't the default, but the radix (lol). Commented.